tests/FakeItEasy.Specs/UnnaturalFakeSpecs.cs

```diff
@@ -159,6 +161,113 @@ public static void AnyCall(Fake<IFoo> fake, bool wasCalled)
+                .x(() => exception
+                        .Should().BeAnExceptionOfType<ArgumentException>()
+                        .And.Message.Should().Be("The target of this call is not the fake object being configured."));
+        }
+
+        [Scenario]
+        public static void CallsToNonVoidWithCallToWrongFake(Fake<IFoo> fake, IFoo wrong, Exception exception)
+        {
+            "Given an unnatural fake"
+                .x(() => fake = new Fake<IFoo>());
+
+            "And an unrelated instance of the faked type"
+                .x(() => wrong = A.Fake<IFoo>());
+
+            "When I configure a fake with an expression that calls the wrong fake"
+                .x(() => exception = Record.Exception(() => fake.CallsTo(f => wrong.MethodWithResult()).Returns(42)));
+
+            "Then it throws an exception that describes the problem"
+                .x(() => exception
+                        .Should().BeAnExceptionOfType<ArgumentException>()
+                        .And.Message.Should().Be("The target of this call is not the fake object being configured."));
+        }
+
+        [Scenario]
+        [Example(int.MinValue)]
+        [Example(-42)]
+        [Example(0)]
+        [Example(42)]
+        [Example(int.MaxValue)]
+        public static void CallsToSetAnyValue(int value, Fake<IFoo> fake, bool wasCalled)
+        {
+            "Given an unnatural fake"
+                .x(() => fake = new Fake<IFoo>());
+
+            "And assignment of a property is configured for any value"
+                .x(() => fake.CallsToSet(f => f.Bar).Invokes(call => wasCalled = true));
+
+            $"When I assign the property to {value}"
+                .x(() => fake.FakedObject.Bar = value);
+
+            "Then the configured behavior is used"
+                .x(() => wasCalled.Should().BeTrue());
+        }
+
+        [Scenario]
+        public static void CallsToSetSpecificValueAndAssigningThatValue(Fake<IFoo> fake, bool wasCalled)
+        {
+            "Given an unnatural fake"
+                .x(() => fake = new Fake<IFoo>());
+
+            "And assignment of a property is configured for a specific value"
+                .x(() => fake.CallsToSet(f => f.Bar).To(42).Invokes(call => wasCalled = true));
+
+            "When I assign the property to that value"
+                .x(() => fake.FakedObject.Bar = 42);
+
+            "Then the configured behavior is used"
+                .x(() => wasCalled.Should().BeTrue());
+        }
+
+        [Scenario]
+        public static void CallsToSetSpecificValueAndAssigningDifferentValue(Fake<IFoo> fake, bool wasCalled)
+        {
+            "Given an unnatural fake"
+                .x(() => fake = new Fake<IFoo>());
+
+            "And assignment of a property is configured for a specific value"
+                .x(() => fake.CallsToSet(f => f.Bar).To(42).Invokes(call => wasCalled = true));
+
+            "When I assign the property to that value"

```

Comment:
> Or maybe "assigning a different value to the property", now that I'm reading the words again.

Yes, that's better. I took the wording from `ConfiguringPropertySetterSpecs` and didn't think too much about it.